src/impl/collection_change_builder.cpp

```diff
@@ -652,3 +653,137 @@ CollectionChangeSet CollectionChangeBuilder::finalize() &&
         std::move(columns)
     };
 }
+
+void ObjectChangeSet::insertions_add(ObjectKeyType obj)
+{
+    m_insertions.insert(obj);
+}
+
+void ObjectChangeSet::modifications_add(ObjectKeyType obj, ColKeyType col)
+{
+    auto it = m_modifications.find(obj);
```

Comment:
This whole function is just `m_modifications[obj].insert(col)`.

Reply:
Good point, I've simplified it a bunch. It just needs to handle the case where the key exists already so that the new column is inserted into the existing set of modified columns instead of wiping and setting the column key set.